### DISCUSSION TITLE
[WIP] Items not set in schema.yaml

### DISCUSSION THREAD

**Discussion by nsano-rururu** [jertel/elastalert2#742] [Ideas]
Update 2022.3.30

----

I'm still investigating, but it seems that the validation check settings for the following items are not defined in schema.yaml.

**TheHive**

No setting

**PagerDuty**

pagerduty_v2_payload_custom_details

**Opsgenie**

opsgenie_recipients
opsgenie_recipients_args
opsgenie_default_receipients
opsgenie_teams
opsgenie_teams_args
opsgenie_default_teams
opsgenie_tags
opsgenie_subject_args

**HTTP POST**

http_post_payload
http_post_static_payload
http_post_headers
http_post_all_values

**HTTP POST 2**

http_post2_payload
http_post2_raw_fields
http_post2_headers
http_post2_all_values

**DingTalk**

dingtalk_btns

**Metric Aggregation**

metric_agg_script

**Spike Aggregation**

metric_agg_script

**Comment by jertel** [comment]
If it's just used here: 

```
    def percentage_violation(self, match_percentage):
        if 'max_percentage' in self.rules and match_percentage > self.rules['max_percentage']:
            return True
        if 'min_percentage' in self.rules and match_percentage < self.rules['min_percentage']:
            return True
        return False
```
Then it appears either integer or decimal would work.

**Reply by nsano-rururu** [reply]
>Then it appears either integer or decimal would work.

thank you for answering. Make the following settings to allow integers and decimal points.

```yaml
min_percentage: {type: number}
max_percentage: {type: number}
```
